src/main/java/vds/domain/policy2/visitlocation/CustomerVisitLocationDemographicBasedCondition.java

```diff
@@ -0,0 +1,73 @@
+package vds.domain.policy2.visitlocation;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import vds.domain.entity.customer.Profile;
+import vds.domain.event.customer.CustomerVisitLocationEvent;
+import vds.domain.policy2.PolicyCondition;
+
+import java.lang.reflect.Field;
+import java.util.*;
+
+@AllArgsConstructor
+public class CustomerVisitLocationDemographicBasedCondition implements PolicyCondition<CustomerVisitLocationEvent> {
+    private final Map<String, Object> props;
+
+    @Getter
+    @AllArgsConstructor
+    private enum Operator {
+        GT(">"),
+        LT("<"),
+        EQ("=");
+
+        private final String value;
+
+        private static final Map<String, Operator> map;
+        static {
+            Map<String, Operator> mx = new HashMap<>();
+            for (Operator op : Operator.values()) {
+                mx.put(op.getValue(), op);
+            }
+            map = Collections.unmodifiableMap(mx);
+        }
+
+        public static Operator fromString(String s) {
+            return map.get(s);
+        }
+    }
+
+    @Override
+    public boolean handle(CustomerVisitLocationEvent data) {
+        List<String> targetAttr = (List<String>) props.get("target");
+        List<Comparable<Object>> values = new ArrayList<>();
+        List<Operator> operators = new ArrayList<>();
+        for (String target : targetAttr) {
+            values.add((Comparable<Object>) props.get(target + "_value"));
```

Comment:
cái này là implicit design! và có lỗ hổng ở usage level
```java
        m2.put("target", Arrays.asList("age"));
        m2.put("age_operator", "<"); 
        m2.put("age_value", 24);
```
=> người ta có thể truyền vào m2.put("a_op", ";") 

=> người ta có thể truyền vào m2.put("a_val", "ok")
chỗ này k phải là 1 Strong design em nhé... nó phải là Explicit... để đảm bảo tạo nên Condition value strong cả trong lẫn ngoài!
https://www.quora.com/What-is-meant-by-implicit-and-explicit-in-programming

Reply:
Đây là đoạn code em form lên mà nó ko nói lên đc gì cả... mà chỉ người code mới hiểu cái nội hàm của nó
```java
        conditions = new ArrayList<>();
        Map<String, Object> m1 = new HashMap<>();
        m1.put("location", "HANOI");
        m1.put("from_hrs", 0);
        m1.put("to_hrs", 12);
```
Tức là người ta có thể truyền 1 đám Map tuỳ biến, vô nghĩa vào trong cái Condition của em...
Nó buộc phải hãm native code... và tăng hàm lượng self-descriptive (tự phát biểu đc tính ứng dụng)